packages/flutter/lib/src/painting/image_resolution.dart

```diff
@@ -240,19 +240,39 @@ class AssetImage extends AssetBundleImageProvider {
     // TODO(ianh): implement support for config.locale, config.textDirection,
     // config.size, config.platform (then document this over in the Image.asset
     // docs)
-    return _findNearest(mapping, config.devicePixelRatio!);
+    return _findNearest(mapping, config.devicePixelRatio!, config.platform);
```

Comment:
nit: remove `config.platform` from the comment above?

nit: maybe just provide `config`?